py/src/pathfinder_worker/call.py

```diff
@@ -968,16 +1051,57 @@ async def do_estimate_fee(
         # there is opposition to providing the non-multiplied scalar value from
         # cairo-lang.
         fees.append(
-            {
-                "gas_consumed": tx_info.actual_fee // max(1, block_info.gas_price),
-                "gas_price": block_info.gas_price,
-                "overall_fee": tx_info.actual_fee,
-            }
+            FeeEstimationInfo(
+                gas_price=block_info.gas_price,
+                gas_usage=tx_info.actual_fee // max(1, block_info.gas_price),
+                overall_fee=tx_info.actual_fee,
+            )
         )
 
     return fees
 
 
+async def do_simulate_tx(
+    async_state: CachedState,
+    general_config: StarknetGeneralConfig,
+    block_info: BlockInfo,
+    transactions: List[AccountTransaction],
+    skip_validate: bool,
+):
+    simulated_transactions = []
+
+    for external_tx in transactions:
+        tx = InternalAccountTransactionForSimulate.create_for_simulate(
+            external_tx, general_config, skip_validate
+        )
+        tx_info = await tx.apply_state_updates(async_state, general_config)
+
```

Comment:
Hmm, looks like this is missing the extra hacks required to handle newly declared classes (that has been added to `do_estimate_fee()`). Maybe we should extract the actual execution part into a function that just returns the list of `tx_info` objects and use that both in `do_estimate_fee()` and `do_simulate_tx()`?